pkg/mcs/meta_storage/server/grpc_service.go

```diff
@@ -75,7 +75,7 @@ func (s *Service) RegisterRESTHandler(userDefineHandlers map[string]http.Handler
 }
 
 func (s *Service) checkServing() error {
-	if !s.manager.srv.IsServing() {
+	if s == nil || s.manager == nil || s.manager.srv == nil || !s.manager.srv.IsServing() {
```

Comment:
Maybe only checking the server is enough?

Reply:
nope, it isn't enough. The reason is s.manager == nil or s.manager.srv == nil. If you remove one of them, you can almost constantly repro the issue by killing pd/api server while keeping sending requests by tools like pd-tso-bench